### PR TITLE
test: update standalone test

### PR DESCRIPTION
**Pull request by himself65**
Upstream: https://github.com/dai-shi/waku/pull/374

Real standalone test:

1. isolated environment
2. install waku by file package instead of copy files

### CODE REVIEW

File: e2e/07_router_standalone.spec.ts [hunk 1]
```diff
@@ -30,69 +30,69 @@ async function testRouterExample(page: Page, port: number) {
...
+      cwd: standaloneDir,
+      env: {
+        ...process.env,
+        PORT: `${port}`,
       },
-    );
+    });
     debugChildProcess(cp);
     await testRouterExample(page, port);
     cp.kill();
```
Comment (himself65) [inline e2e/07_router_standalone.spec.ts:98]: I guess cp.kill won't really kill a waku process

**Comment by himself65** [comment]
I dont know why!!!

File: e2e/07_router_standalone.spec.ts
Comment (himself65) [inline e2e/07_router_standalone.spec.ts:1]: I want to change this test to real standalone but failed

**Comment by Aslemammad** [comment]
@himself65 If you still have issue with this I can take a look later, if yes, then I'd be happy if @dai-shi would assign me here.

**Comment by himself65** [comment]
> @himself65 If you still have issue with this I can take a look later, if yes, then I'd be happy if @dai-shi would assign me here.

If you revert this commit you will see https://github.com/dai-shi/waku/pull/420/commits/af47f13f828e02777cc37817f5a55229aed12b59

**Comment by himself65** [comment]
Let me reproduce this step by step🤔

**Comment by himself65** [comment]
@Aslemammad yes we can see

**Comment by Aslemammad** [comment]
@himself65 can we rebase this first so I can see the issue with newer results? because apparently the counter example is removed.

**Comment by dai-shi** [comment]
@Aslemammad If you like, you can fork and open a new PR.

**Comment by himself65** [comment]
rebased

File: packages/waku/src/lib/handlers/handler-dev.ts [hunk 2]
```diff
@@ -154,12 +154,17 @@ export function createHandler<
...
+          config.basePath,
+          config.srcDir,
+          config.mainJs,
+        );
         const readable = await renderHtml({
           config,
           pathname: req.url.pathname,
           searchParams: req.url.searchParams,
           htmlHead: `${config.htmlHead}
-<script src="${config.basePath}${config.srcDir}/${config.mainJs}" async type="module"></script>`,
```
Comment (dai-shi) [inline packages/waku/src/lib/handlers/handler-dev.ts:162]: Was this causing any problem? `joinPath` isn't specifically designed for URLs.

Comment (himself65) [reply packages/waku/src/lib/handlers/handler-dev.ts]: No, I just wanna saw if this cause the error

**Comment by dai-shi** [comment]
https://github.com/dai-shi/waku/pull/471/files#r1485443331
> shouldn't we also test `--with-ssr`?

**Comment by dai-shi** [comment]
I'm looking forward to this very much. Please make it ready for review when the issue is resolved.

**Comment by himself65** [comment]
let me do this this weekend 💪

**Review by himself65 — COMMENTED** [review]
I guess everything works again now 🙏 🙏 🙏

**Review by himself65 — COMMENTED** [review]
Also bump the vite version because I thought it was related, but not.

File: package.json [hunk 3]
```diff
@@ -73,13 +74,14 @@
...
     "react-server-dom-webpack": "18.3.0-canary-14fd9630e-20240213",
+    "terminate": "^2.6.1",
     "typescript": "^5.3.3",
     "wait-port": "^1.1.0",
     "waku": "workspace:*"
   },
   "pnpm": {
     "overrides": {
-      "vite": "5.0.12"
+      "vite": "5.1.3"
```
Comment (dai-shi) [inline package.json:84]: Oh, we forgot to update this in #485. @Aslemammad

**Review by dai-shi — APPROVED** [review]
Thanks so much for you effort!

File: e2e/07_router_standalone.spec.ts [hunk 4]
```diff
@@ -36,73 +38,83 @@ async function testRouterExample(page: Page, port: number) {
...
-      stdio: 'inherit',
+
+    // copy again?
+    await cp(exampleDir, standaloneDir, {
+      filter: (src) => {
+        return src.includes('src');
+      },
+      force: true,
+      recursive: true,
     });
```
Comment (himself65) [inline e2e/07_router_standalone.spec.ts:73]: Oh I should remove this

**Pull request merged**